shell/platform/glfw/client_wrapper/include/flutter/flutter_window.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_GLFW_CLIENT_WRAPPER_INCLUDE_FLUTTER_FLUTTER_WINDOW_H_
+#define FLUTTER_SHELL_PLATFORM_GLFW_CLIENT_WRAPPER_INCLUDE_FLUTTER_FLUTTER_WINDOW_H_
+
+#include <string>
+#include <vector>
+
+#include <flutter_glfw.h>
+
+#include "plugin_registrar.h"
+
+namespace flutter {
+
+// A window displaying Flutter content.
+class FlutterWindow {
+ public:
+  explicit FlutterWindow(FlutterDesktopWindowRef window) : window_(window) {}
+
+  ~FlutterWindow() = default;
+
+  // Prevent copying.
+  FlutterWindow(FlutterWindow const&) = delete;
+  FlutterWindow& operator=(FlutterWindow const&) = delete;
+
+  // Enables or disables hover tracking.
+  //
+  // If hover is enabled, mouse movement will send hover events to the Flutter
+  // engine, rather than only tracking the mouse while the button is pressed.
+  // Defaults to off.
+  void SetHoverEnabled(bool enabled) {
+    FlutterDesktopWindowSetHoverEnabled(window_, enabled);
+  }
+
+  // Sets the displayed title of the window.
+  void SetTitle(const std::string& title) {
+    FlutterDesktopWindowSetTitle(window_, title.c_str());
+  }
+
+  // Sets the displayed icon for the window.
+  //
+  // The pixel format is 32-bit RGBA. The provided image data only needs to be
+  // valid for the duration of the call to this method. Pass a nullptr to revert
+  // to the default icon.
+  void SetIcon(uint8_t* pixel_data, int width, int height) {
+    FlutterDesktopWindowSetIcon(window_, pixel_data, width, height);
+  }
+
+ private:
+  // Handle for interacting with the C API's window.
+  FlutterDesktopWindowRef window_;
```

Comment:
I am unclear about how a window that is created but not consumed by FlutterDesktopRunWindowLoop is collected.

Reply:
Maybe have a static method that takes a smart pointer to the FlutterWindow and consumes the same.

```
static void FlutterWindow::Run(std::unique_ptr<FlutterWindow>)
```

In the dtor, collect the window ref if the windows runloop was never run.

Reply:
This is a preexisting issue with the C API; you can't not run the loop without leaking (see https://github.com/flutter/engine/blob/master/shell/platform/glfw/public/flutter_glfw.h#L50). There's no underlying method that can be called in the destructor at the moment, just as there wasn't when this code was part of `FlutterViewController`.

I was going to fix that when I reworked the runloop handling in the GLFW shell in general (there needs to be a way to do incremental loop runs, e.g., to allow having another window using a different runloop), but I can add just the stand-alone destruction method now.

Reply:
sgtm

Reply:
Done. The GLFW API definitely still has structural issues that need to be resolved, but that's another step in the right direction.